src/sqlfluff/rules/L018.py

```diff
@@ -117,10 +117,7 @@ def indent_size_up_to(segs):
                             # Is it all whitespace before the bracket on this line?
                             prev_segs_on_line = [
                                 elem
-                                for elem in context.segment.iter_segments(
-                                    expanding=["common_table_expression", "bracketed"],
-                                    pass_through=True,
-                                )
+                                for elem in context.segment.raw_segments
```

Comment:
This is a fascinating bug. Apparently, the rule was designed to handle this case, but:
* There was no test for it.
* It didn't work.
* Apparently, the `else:` block starting on line 134 was never executing. At least, not in any of the `L018.yml` test cases.